monai/apps/deepgrow/handler.py

```diff
@@ -0,0 +1,287 @@
+                d = ImageDraw.Draw(ti)
+                t = "region: {}".format(region)
+                if self.compute_metric:
+                    t = t + "\ndice: {:.4f}".format(metric.mean())
+                    t = t + "\nstdev: {:.4f}".format(metric.stdev())
+                d.multiline_text((10, 10), t, fill=(255, 255, 0))
+                ti = rescale_array(np.rollaxis(np.array(ti), 2, 0)[0][np.newaxis])
+                all_imgs.append(ti)
+            all_imgs.extend(region_data)
+
+        if len(all_imgs[0].shape) == 3:
+            img_tensor = make_grid(tensor=torch.from_numpy(np.array(all_imgs)), nrow=4, normalize=True, pad_value=2)
+            self.writer.add_image(tag=f"Deepgrow Regions ({self.tag_name})", img_tensor=img_tensor, global_step=epoch)
+
+        if len(all_imgs[0].shape) == 4:
+            for region in sorted(self.plot_data.keys()):
+                tags = [f"region_{region}_image", f"region_{region}_label", f"region_{region}_output"]
+                if torch.distributed.is_initialized():
+                    rank = "r{}-".format(torch.distributed.get_rank())
+                    tags = [rank + tags[0], rank + tags[1], rank + tags[2]]
+                for i in range(3):
+                    img = self.plot_data[region][i]
+                    img = np.moveaxis(img, -3, -1)
+                    plot_2d_or_3d_image(
+                        img[np.newaxis], epoch, self.writer, 0, self.max_channels, self.max_frames, tags[i]
+                    )
+
+        self.logger.info(
+            "Saved {} Regions {} into Tensorboard at epoch: {}".format(
+                len(self.plot_data), sorted([*self.plot_data]), epoch
+            )
+        )
+        self.writer.flush()
+
+    def write_region_metrics(self, epoch):
+        metric_sum = 0
+        means = {}
+        for region in self.metric_data:
+            metric = self.metric_data[region].mean()
+            self.logger.info(
+                "Epoch[{}] Metrics -- Region: {:0>2d}, {}: {:.4f}".format(epoch, region, self.tag_name, metric)
+            )
+
+            if self.merge_scalar:
+                means["{:0>2d}".format(region)] = metric
+            else:
+                self.writer.add_scalar("{}_{:0>2d}".format(self.tag_name, region), metric, epoch)
+            metric_sum += metric
+
+        if self.merge_scalar:
+            means["avg"] = metric_sum / len(self.metric_data)
+            self.writer.add_scalars("{}_region".format(self.tag_name), means, epoch)
+        elif len(self.metric_data) > 1:
+            metric_avg = metric_sum / len(self.metric_data)
+            self.writer.add_scalar("{}_regions_avg".format(self.tag_name), metric_avg, epoch)
+        self.writer.flush()
+
+    def __call__(self, engine: Engine, action) -> None:
+        total_steps = engine.state.iteration
+        if total_steps < engine.state.epoch_length:
+            total_steps = engine.state.epoch_length * (engine.state.epoch - 1) + total_steps
```

Comment:
my understanding is that total_steps is the total number of iterations.
total_steps is iteration if we are in the very first epoch,
otherwise `total_steps = engine.state.iteration + engine.state_epoch_length * (engine.state.epoch - 1)`
should this condition be
`if engine.state.epoch > 0`?

Reply:
removed this code.. not need as part of core functionality for deepgrow